src/clnt_generic.c

```diff
@@ -517,11 +503,12 @@ clnt_req_setup(struct clnt_req *cc, struct timespec timeout)
 		__warnx(TIRPC_DEBUG_FLAG_ERROR,
 			"%s: %p fd %d insert failed xid %" PRIu32,
 			__func__, &rec->xprt, rec->xprt.xp_fd, cc->cc_xid);
-		return (false);
+		cc->cc_error.re_status = RPC_TLIERROR;
+		return (RPC_TLIERROR);
 	}
 
 	CLNT_REF(clnt, CLNT_REF_FLAG_NONE);
-	return (true);
+	return (RPC_SUCCESS);
```

Comment:
This path doesn't set cc_error.  RPC_SUCCESS is zero, so it should be okay, but in the function above, you explicitly set cc_error in the success case.  We should probably stick to one form or the other.